scripts/prepare_sector_network.py

```diff
@@ -1455,113 +1455,264 @@ def add_land_transport(n, costs):
+        + cycling_shift(transport[nodes], 2)
+    ) / 3
+    n.madd(
+        "Load",
+        nodes,
+        suffix=" land transport",
+        bus=nodes + " land transport",
+        carrier="land transport demand",
+        p_set=p_set,
+    )
+
+    # Add EV links
+    if electric_share != 0:
         n.add("Carrier", "Li ion")
 
         n.madd(
             "Bus",
             nodes,
             location=nodes,
-            suffix=" EV battery",
+            suffix=" EV driving",
             carrier="Li ion",
             unit="MWh_el",
         )
 
-        p_set = (
-            electric_share
-            * (
-                transport[nodes]
-                + cycling_shift(transport[nodes], 1)
-                + cycling_shift(transport[nodes], 2)
-            )
-            / 3
-        )
-
-        n.madd(
-            "Load",
-            nodes,
-            suffix=" land transport EV",
-            bus=nodes + " EV battery",
-            carrier="land transport EV",
-            p_set=p_set,
-        )
-
-        p_nom = number_cars * options.get("bev_charge_rate", 0.011) * electric_share
-
         n.madd(
             "Link",
             nodes,
             suffix=" BEV charger",
             bus0=nodes,
-            bus1=nodes + " EV battery",
-            p_nom=p_nom,
+            bus1=nodes + " EV driving",
+            p_nom_extendable=True,
             carrier="BEV charger",
             p_max_pu=avail_profile[nodes],
-            efficiency=options.get("bev_charge_efficiency", 0.9),
+            efficiency=1,  # instead of options.get("bev_charge_efficiency", 0.9) -> efficiency already accounted for in build_transport_demand
             # These were set non-zero to find LU infeasibility when availability = 0.25
             # p_nom_extendable=True,
             # p_nom_min=p_nom,
             # capital_cost=1e6,  #i.e. so high it only gets built where necessary
         )
 
-    if electric_share > 0 and options["v2g"]:
+        # Only when v2g option is True, add the v2g link
+        if options["v2g"]:
+            n.madd(
+                "Link",
+                nodes,
+                suffix=" V2G",
+                bus1=nodes,
+                bus0=nodes + " EV driving",
+                p_nom_extendable=True,
+                carrier="V2G",
+                p_max_pu=avail_profile[nodes],
+                efficiency=options.get("bev_charge_efficiency", 0.9),
+            )
+
+        # add bev managment when enabled
+        n.add("Carrier", "EV battery storage")
+
+        if options["bev_dsm"]:
+            n.madd(
+                "Store",
+                nodes,
+                suffix=" EV battery storage",
+                bus=nodes + " EV driving",
+                carrier="EV battery storage",
+                e_cyclic=True,
+                e_nom_extendable=True,
+                e_max_pu=1,
+                e_min_pu=dsm_profile[nodes],
+                # lifetime?
```

Comment:
I set the battery lifetime to the car lifetime in commit 239bee993d928bfd9c0fcd58d185cd3055d26430 while I was changing the constraints in the solve network to prevent them being used as v2g while the land transport EVs are already decomissioned, but we can discuss this